### PR TITLE
Banner story with typescript

### PR DESCRIPTION
**Pull request by wenche**
Part of #735

### CODE REVIEW

**Comment by wenche** [comment]
Stories with multiple components are a [separate thing](https://storybook.js.org/docs/react/workflows/stories-for-multiple-components) with Storybook. The concept of control doesn't work for children.

**Comment by wenche** [comment]
Also, I think the Banner is a candidate for some alignment.. It has children as explicit props atm

**Review by pomfrida — APPROVED** [review]
LGTM 👍

File: libraries/core-react/src/Banner/index.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
-const Banner = BaseBanner as BannerProps
+const Banner = BaseBanner as AllBannerProps
 
 Banner.BannerIcon = BannerIcon
 Banner.BannerMessage = BannerMessage
 Banner.BannerActions = BannerActions
 
-export { Banner }
+export { Banner, BannerProps }
```
Comment (vnys) [inline libraries/core-react/src/Banner/index.ts:18]: Shouldn’t the `AllBannerProps` type be exported here? But as `BannerProps`? 🤔 So maybe import `BannerProps as BaseBannerProps`, and then rename `AllBannerProps` to `BannerProps`? Or am I wrong?

**Review by vnys — CHANGES_REQUESTED** [review]
Should have been a “maybe request changes" option here

File: libraries/core-react/src/Banner/index.ts (hunk 1 above)
Comment (vnys) [reply libraries/core-react/src/Banner/index.ts:18]: Hard to know with these compound components… But the components has children, and those children have to be typed, and then I assume that compound type is the type for the component. This is a bit easier to grasp when this is done in the actual component and not when re-exporting. 

```tsx
type ComponentProps = {
  /** Some other colour */
  color?: string
  /** Some foo */
  foo?: string
  /** @ignore */
  children: React.ReactNode
}

type CompoundComponent = React.ForwardRefExoticComponent<ComponentProps> & {
  OtherComponent: typeof OtherComponent
}

/**
 * This is the Component description
 */

const Component = forwardRef<HTMLDivElement, ComponentProps>(function Component(
  { children, color = 'violet' },
  ref,
) {
  return (
    <div style={{ color }} ref={ref}>
      {children}
    </div>
  )
}) as CompoundComponent

Component.OtherComponent = OtherComponent
```

…except the props disappeared from the Storybook when doing it that way…

Comment (wenche) [reply libraries/core-react/src/Banner/index.ts:18]: The AllBannerProps was just a creative rename since it previously was named BannerProps 😅

Comment (wenche) [reply libraries/core-react/src/Banner/index.ts:18]: But yes, we need to figure out what is the best way of reexporting both types and props with compound components

**Comment by wenche** [comment]
Tried to export with * in the outer `index.ts `file to get rid of annoying warning. I had to combine it with a separate type export in Banner `index.ts` to make it work

**Review by pomfrida — APPROVED** [review]
LGTM 👍

**Comment by vnys** [comment]
> Tried to export with * in the outer `index.ts `file to get rid of annoying warning. I had to combine it with a separate type export in Banner `index.ts` to make it work

That’s right, the `*` will just re-export whatever is already exported.

File: libraries/core-react/src/Banner/index.ts [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 
-type BannerProps = typeof BaseBanner & {
+type AllBannerProps = typeof BaseBanner & {
   BannerIcon: typeof BannerIcon
   BannerMessage: typeof BannerMessage
   BannerActions: typeof BannerActions
 }
 
-const Banner = BaseBanner as BannerProps
+const Banner = BaseBanner as AllBannerProps
```
Comment (vnys) [inline libraries/core-react/src/Banner/index.ts:12]:
```suggestion
const Banner = BaseBanner as BannerProps
```

File: libraries/core-react/src/Banner/index.ts [hunk 3]
```diff
@@ -1,18 +1,19 @@
-import { Banner as BaseBanner } from './Banner'
+import { Banner as BaseBanner, BannerProps as Props } from './Banner'
 import { BannerIcon } from './BannerIcon'
 import { BannerMessage } from './BannerMessage'
 import { BannerActions } from './BannerActions'
 
-type BannerProps = typeof BaseBanner & {
+type AllBannerProps = typeof BaseBanner & {
```
Comment (vnys) [inline libraries/core-react/src/Banner/index.ts:6]:
```suggestion
type BannerProps = typeof BaseBanner & {
```

File: libraries/core-react/src/Banner/index.ts [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
 
-const Banner = BaseBanner as BannerProps
+const Banner = BaseBanner as AllBannerProps
 
 Banner.BannerIcon = BannerIcon
 Banner.BannerMessage = BannerMessage
 Banner.BannerActions = BannerActions
 
 export { Banner }
+export type BannerProps = Props
```
Comment (vnys) [inline libraries/core-react/src/Banner/index.ts:19]:
```suggestion
export type BannerProps
```

**Review by vnys — CHANGES_REQUESTED** [review]
I think we should re-export the compound type, right?

**Comment by wenche** [comment]
@vnys Not sure if I understood you correctly, but like this?

**Comment by vnys** [comment]
> @vnys Not sure if I understood you correctly, but like this?

I think you could have just accepted the suggestions in Github and they would have been committed

**Comment by vnys** [comment]
…but yes, that’s the way I think it must be done 😊

**Review by vnys — APPROVED** [review]
LGTM 👍

**Comment by wenche** [comment]
> > @vnys Not sure if I understood you correctly, but like this?
> 
> I think you could have just accepted the suggestions in Github and they would have been committed

I was thinking of that, but I did a couple of things more than you did :/ With the initial import at least

**Pull request merged**